sim.go

```
@@ -1768,18 +1795,34 @@ func (s *Sim) spawnAircraft() {
 			s.lg.Errorf("%s: couldn't find an active runway for spawning departure?", airport)
 			continue
 		}
-
-		prevDep := s.lastDeparture[airport][runway][category]
-		s.lg.Infof("%s/%s/%s: previous departure", airport, runway, category)
-		ac, dep, err := s.World.CreateDeparture(airport, runway, category,
-			s.LaunchConfig.DepartureChallenge, prevDep)
-		if err != nil {
-			s.lg.Errorf("CreateDeparture error: %v", err)
-		} else {
-			s.lastDeparture[airport][runway][category] = dep
-			s.lg.Infof("%s/%s/%s: launch departure", airport, runway, category)
-			s.launchAircraftNoLock(*ac)
-			s.NextDepartureSpawn[airport] = now.Add(randomWait(rateSum, false))
+		for { // There will always be a gate, because is all gates are checked this func wont get called
```

From before I thought that this wasn't true? Wasn't there a case something like if the scenario has 0 departures to some gate then if every other gate but that one is stopped but that one isn't, then it still won't be able to launch any aircraft?

If so, would it work to just continue to call `spawnAircraft()`, let it fail if the gate it chose is stopped, and then assume it will eventually find a gate that's open (if there is one) in the next few times...?

---

Do you mean if there are no departures made/ coded in for that one gate?

---

No, I mean for example, say I launch the "KEWR 4L Departure" scenario and set an ADR of 0 for KMMU and KTEB when I create the sim. Then if I halt departures from KEWR 4L I think this will go into an infinite loop because yes, there are some gates that aren't stopped, but on the other hand a departure from them will never be launched.

---

I tried to reproduce this and looks like it was fixed in [0eb5aec](https://github.com/mmp/vice/pull/30/commits/0eb5aec6726852ee1deccc27c8499c684668ca84).